File: 1544-count-good-nodes-in-binary-tree/count-good-nodes-in-binary-tree.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */
class Solution {
public:
    void solve(TreeNode* root, int &ans, int m){
        if(!root) return;
        // int a;
        if(root->val >= m){
            // a = m;
            m = root->val;
            ans++;
        }
        solve(root->left, ans, m);
        // m = a;
        solve(root->right, ans, m);
    }
    int goodNodes(TreeNode* root) {
        int ans = 0;
        int m = root->val;
        solve(root, ans, m);
        return ans;
    }
};

File: 1544-count-good-nodes-in-binary-tree/count-good-nodes-in-binary-tree.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
class Solution {
    int ans;
    public void solve(TreeNode root, int m){
        if(root == null) return;

        if(root.val >= m) {
            this.ans++;
            m = Math.max(m, root.val);
        }

        solve(root.left, m);

        solve(root.right, m);
    }
    public int goodNodes(TreeNode root) {

        int m = root.val;

        this.ans = 0;

        solve(root, m);

        return ans;
    }
}